File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/fragments/list/ItemComparator.java
package pl.udu.uwr.pum.shoppyjava.fragments.list;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import pl.udu.uwr.pum.shoppyjava.model.Item;

public class ItemComparator extends DiffUtil.ItemCallback<Item> {
    @Override
    public boolean areItemsTheSame(@NonNull Item oldItem, @NonNull Item newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(@NonNull Item oldItem, @NonNull Item newItem) {
        return oldItem.getName().equals(newItem.getName()) &&
                oldItem.getQuantity() == newItem.getQuantity();
    }
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/fragments/list/ItemComparator.kt
package pl.udu.uwr.pum.shoppykotlin.fragments.list

import androidx.recyclerview.widget.DiffUtil
import pl.udu.uwr.pum.shoppykotlin.model.Item

class ItemComparator : DiffUtil.ItemCallback<Item>() {
    override fun areItemsTheSame(oldItem: Item, newItem: Item): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: Item, newItem: Item): Boolean {
        return oldItem.name == newItem.name && oldItem.quantity == newItem.quantity
    }
}